### PR TITLE
Implement `GTF` instruction

### PR DESCRIPTION
**Pull request by vlopes11**
Implements the `GTF` instruction.

The tests are being developed in parallel. This PR is released so the accumulated breaking changes can be unblocked for `fuel-core`.

The tests are expected to be added only incrementally, so they can be released as patch (with eventual bugfixes, if applicable)

### CODE REVIEW

**Comment by vlopes11** [comment]
Blocked by https://github.com/FuelLabs/fuel-tx/pull/169

File: tests/metadata.rs [hunk 1]
```diff
@@ -154,3 +156,84 @@ fn metadata() {
...
+
+    #[rustfmt::skip]
+    let cases = vec![
+        inputs_bytes[0].clone(),
+    ];
+
+    #[rustfmt::skip]
+    let script = vec![
+        Opcode::MOVI(0x20, 0x01),
+        Opcode::gtf(0x30, 0x00, GTFArgs::ScriptData),
```
Comment (Dentosal) [inline tests/metadata.rs:196]: Why is gtf here lowercase?

**Review by Dentosal — CHANGES_REQUESTED** [review]

File: tests/metadata.rs (hunk 1 above)
Comment (vlopes11) [reply tests/metadata.rs:196]: We should probably split and build a dedicated dsl for the tx scripts, but its currently centralized in `Opcode` that that one is a function: [Opcode::gtf](https://docs.rs/fuel-asm/latest/fuel_asm/enum.Opcode.html#method.gtf)

Comment (Dentosal) [reply tests/metadata.rs:196]: Oh so that's just for converting from GTFArgs to Imm12

**Review by Dentosal — APPROVED** [review]

**Pull request merged**